python/pyspark/sql/types.py

```diff
@@ -442,7 +442,7 @@ def needConversion(self) -> bool:
 
     def toInternal(self, dt: datetime.timedelta) -> Optional[int]:
         if dt is not None:
-            return (math.floor(dt.total_seconds()) * 1000000) + dt.microseconds
+            return (((dt.days * 86400) + dt.seconds) * 1_000_000) + dt.microseconds
```

Comment:
nit, is `1_000_000` supported in all 3.7+ versions?

Reply:
Oh good point. it should be supported from 3.6+ https://peps.python.org/pep-0515/